x-pack/legacy/plugins/security/server/routes/api/v1/authenticate.js

```diff
@@ -82,6 +83,36 @@ export function initAuthenticateApi({ authc: { login, logout }, config }, server
     }
   });
 
+  /**
+   * The route should be configured as a redirect URI in OP when OpenID Connect implicit flow
+   * is used, so that we can extract authentication response from URL fragment and send it to
+   * the `/api/security/v1/oidc` route.
+   */
+  server.route({
```

Comment:
__note:__ didn't write unit tests for this endpoint as it's covered by integration tests and we'll need to rewrite them soon in Jest and TypeScript anyway (when we migrate to NP routes).